src/adapter/src/coord/ddl.rs

```diff
@@ -321,62 +338,47 @@ impl<S: Append + 'static> Coordinator<S> {
+        connection: StorageSinkConnection,
         session: Option<&Session>,
     ) -> Result<(), AdapterError> {
         // Update catalog entry with sink connection.
         let entry = self.catalog.get_entry(&id);
         let name = entry.name().clone();
-        let mut sink = match entry.item() {
-            CatalogItem::Sink(sink) => sink.clone(),
+        let sink = match entry.item() {
+            CatalogItem::Sink(sink) => sink,
             _ => unreachable!(),
         };
-        sink.connection = catalog::SinkConnectionState::Ready(connection.clone());
-        // We don't try to linearize the as of for the sink; we just pick the
-        // least valid read timestamp. If users want linearizability across
-        // Materialize and their sink, they'll need to reason about the
-        // timestamps we emit anyway, so might as emit as much historical detail
-        // as we possibly can.
-        let id_bundle = self
-            .index_oracle(compute_instance)
-            .sufficient_collections(&[sink.from]);
-        let frontier = self.least_valid_read(&id_bundle);
-        let as_of = SinkAsOf {
-            frontier,
-            strict: !sink.with_snapshot,
+        let sink = catalog::Sink {
+            connection: StorageSinkConnectionState::Ready(connection.clone()),
+            ..sink.clone()
         };
+
         let ops = vec![
             catalog::Op::DropItem(id),
             catalog::Op::CreateItem {
                 id,
                 oid,
-                name: name.clone(),
+                name,
                 item: CatalogItem::Sink(sink.clone()),
             },
         ];
-        let df = self
-            .catalog_transact(session, ops, |txn| {
-                let mut builder = txn.dataflow_builder(compute_instance);
-                let from_entry = builder.catalog.get_entry(&sink.from);
-                let sink_description = mz_storage::types::sinks::SinkDesc {
-                    from: sink.from,
-                    from_desc: from_entry
-                        .desc(
-                            &builder
-                                .catalog
-                                .resolve_full_name(from_entry.name(), from_entry.conn_id()),
-                        )
-                        .unwrap()
-                        .into_owned(),
-                    connection: connection.clone(),
-                    envelope: Some(sink.envelope),
-                    as_of,
-                };
-                Ok(builder.build_sink_dataflow(name.to_string(), id, sink_description)?)
-            })
-            .await?;
 
-        Ok(self.ship_dataflow(df, compute_instance).await)
+        let () = self.catalog_transact(session, ops, move |_| Ok(())).await?;
+        // XXX(chae): should this happen in the same task where we build the connection??  Or should it need to happen
+        // after we update the catalog?
+        match self.create_storage_export(id, &sink, connection).await {
```

Comment:
The general pattern we use is that once `catalog_transact` completes sucessfully then we just call `unwrap()` on the STORAGE/COMPUTE controller methods. For example: https://github.com/MaterializeInc/materialize/blob/8dc2cc9fe6533a9d69108f9d0f4cf19c91674512/src/adapter/src/coord/sequencer.rs#L426-L473

Is there any specific reason that we'd expect `create_storage_export` to fail other than STORAGE being unavailable?

Reply:
Is it in case `sink.from` was deleted while we're waiting for the sink to connect?

Reply:
Yeah I believe that's something that could happen.  I guess it's a question of whether it's worse to (for the time being):
- panic if there's an error
- have a non-transactional Create / error / cleanup